types/resource.go

```diff
@@ -67,6 +67,7 @@ type VolumeSpec struct {
 	DiskSelector        []string       `json:"diskSelector"`
 	NodeSelector        []string       `json:"nodeSelector"`
 	DisableFrontend     bool           `json:"disableFrontend"`
+	DisableExpansion    bool           `json:"disableExpansion"`
```

Comment:
Can user set the volume.spec.Size to the original size instead? It should still in the engine or volume status since it's not done yet.

Then we don't need this field and extra API call.

Reply:
No. When `v.Spec.Size == e.Status.CurrentSize`, Longhorn needs to distinguish if the engine expansion is already complete or just canceled by users.

Reply:
Engine hasn't finish updated yet, why `e.Status.CurrentSize` is updated already?

Reply:
How about if expanding in progress, then that's cancel. Otherwise it's done.

Reply:
If expanding (enigne) is in progress: `v.Spec.Size == e.Spec.Size > e.Status.CurrentSize`
If expansion is canceled: `v.Spec.Size == e.Status.CurrentSize < e.Spec.Size`.

It's possible to remove this field, but I am not sure if there can be some other cases leading to this situation, or if it's too tricky.

Reply:
The fields in the definition should describe state, not action. This field is an action instead of a state. It shouldn't be in the definition anyway.